ncm-afsclt/src/main/perl/afsclt.pm

```diff
@@ -2,997 +2,306 @@
-
-  if ($config->elementExists($mypath."/cellservdb")) {
-    $master_cellservdb_cdb = $config->getValue($mypath."/cellservdb");
-
-    if ( $master_cellservdb_cdb =~ m;^(file://)?(\/[\w\/.-]+);i) { # absolute file name, alphanum, slashes, -, _
-      if ( -r "$2" ) {
-	$master_cellservdb = "file://$2";
-      } else {
-	$self->warn("cannot access cellservdb $2, giving up");
-	return 0;
-      }
-    } elsif ( $master_cellservdb_cdb =~ m;^((ftp|http|file)://.+);i) { # known URIs - could contain other stuff i.e. for programmatic access
-      $master_cellservdb = $1;
-    } else {
-      $self->warn("weird/unsafe cellservdb URL/PATH: $master_cellservdb, giving up");
-      return 0;
-    }
-  } else {
-    $self->warn("cannot get CDB entry $mypath/cellservdb, cannot keep CellServDB updated");
-    return 0;
-  }
-  # LWP::Simple means  no error messages etc.
-  $self->info("will get cellservdb information from $master_cellservdb");
-
-  # FIXME: UGLY HACK here - it appears that one can not "untaint" the
-  # location variable via the above, any reference to $1 and such still leaves
-  # it somehow tainted. Only explicit assignment seems to remove this.
-  # The below is supposed to "cure" the following messages on SLC4, perl-5.8.8-10.el5_0.2, perl-libwww-perl-5.79-5
-  ## Insecure dependency in connect while running with -t switch at /usr/lib/perl5/5.8.8/i386-linux-thread-multi/IO/Socket.pm line 114.
-  ## Insecure dependency in connect while running with -t switch at /usr/lib/perl5/5.8.8/i386-linux-thread-multi/IO/Socket.pm line 120.
-
-  my $ugly_hack;
-  if ($master_cellservdb =~ m;(http://consult.cern.ch/service/afs/CellServDB);) {
-      $ugly_hack = 'http://consult.cern.ch/service/afs/CellServDB';
-      $master_cellservdb = $ugly_hack;
-  }
-  # end of UGLY HACK.
-
-  my $cellservdb_content = LWP::Simple::get($master_cellservdb);
-  if (! $cellservdb_content) {
-    $self->info("cannot read cellservdb $master_cellservdb, AFS cell info not changed");
     return 1;
-  }
-
-  if ($NoAction) {
-    $self->info("skipping CellServDB check and in-kernel update");
-  } else {
-    LC::Check::file("CellServDB",
-		    "contents"    => $cellservdb_content,
-		    "destination" => $localcelldb,
-		    "owner"       => 0,
-		    "mode"        => 0444,
-		   );
-    $self->update_afs_cells();
-  }
-
-  return 1;
 }
 
-# This code comes from sue/feature/afs/update.pl, slightly modified
-# to work in our context (replace afsutil with "fs")
-#
 # update the list of known AFS cells and run "fs newcell" when needed
 # This will only add new cells, not remove currently known ones.
 #
 sub update_afs_cells ( $$ ) {
-    my($self) = @_;
-    my($cell, %seen, %ipaddrs, @hosts, $todo, @todo, $error, $afsutil);
-    local(*FH, $_);
-    # init
-
-    # read CellServDB
-    # take it from the real file
-    if( ! open(*FH, $localcelldb)) {
-      $self->warn("cannot read local $localcelldb, giving up");
-      return 0;
-    }
+    my ($self) = @_;
```

Comment:
i think this method is `NoAction` safe on it's own. no need for the `if` construct above